dspace-api/src/main/java/org/orcid/jaxb/model/clientgroup/ClientType.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * =============================================================================
+ *
+ * ORCID (R) Open Source
+ * http://orcid.org
+ *
+ * Copyright (c) 2012-2014 ORCID, Inc.
+ * Licensed under an MIT-Style License (MIT)
+ * http://orcid.org/open-source-license
```

Comment:
I'd like to better understand why we are redistributing ORCID source code in DSpace.  Is this code not available via Maven?  Does ORCID provide this code in any other (precompiled) form?

Reply:
@tdonohue 
I wholeheartedly agree that this should either be contained in a separate maven dependency or automatically generated as this just opens up the possibility for accidental changes to the code (With preference for the separate maven dependency, which would require less changes when an update is needed-> Since these schemes are contained in several different xsd's)

I already mentioned this in a comment on slack: 
I had contacted feedback/support with this idea and it seems they'll take this into consideration for the future.
https://support.orcid.org/forums/175591-orcid-ideas-forum/suggestions/18563095-a-maven-dependency-would-be-created-containing-the
Perhaps if we can mention that we are planning to put this into a separate dependency ourselves, this might give them some incentive to deploy it themselves, they might want to reduce the number of diverging codebases.

Reply:
Sorry to jump in the conversation, in DSpace-CRIS (we released support for ORCID V.2 from September 2017 see https://www.4science.it/en/2017/09/14/dspace-cris-5-8-and-orcid-v2-api-support-release-announcement/) we using an external mvn repo to manage this dependency. See https://github.com/4Science/DSpace/blob/dspace-6_x_x-cris/pom.xml#L1679 and https://github.com/4Science/mvn-repo/tree/master/releases/it/4science/tools-orcid-jaxb/2.0 

Maybe, simply you can try to use this dependency from github 4Science maven repository (I think that should be work also for 2.1 version of API) or following the same approach, create a new git repo and release all the jaxb artifact (as @tdonohue said before) to the maven repo.

Reply:
@tdonohue just a quick reminder we discussed this beforehand in https://jira.duraspace.org/browse/DS-3447?focusedCommentId=58630&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-58630

This has been delivered based on those specs

Reply:
Perhaps we can find a way to delicately let ORCID discover that we would help them get an artifact out, if that would make it happen more expeditiously.

Reply:
@benbosman : I recall this was previously discussed as a problem, but my expectations were that we'd not embed this code into our API layer. I see that was never clarified in our previous discussion.  I think, at a minimum, we need this to be a separate GitHub repo, with clear documentation on how these Java sources were generated (so that we can regenerate them in the future, as necessary). 

We also could look closer at how @lap82 noted DSpace-CRIS supports ORCIDv2.

If I could have clear documentation on how these `org.orcid.*` source files were generated (perhaps attach it to the ticket?), I could chip in here and potentially generate them in a separate GitHub repo. I'll also add a comment to ORCID about this -- thanks @jonas-atmire for the link.

Reply:
@lap82 : Is the source code behind this 4Science Maven repo available somewhere? https://github.com/4Science/mvn-repo/tree/master/releases/it/4science/tools-orcid-jaxb/2.0  I'd like to consider recreating this as a central DSpace GitHub project & releasing to Maven Central. That way DSpace-CRIS could also use that dependency.

It looks to me like you've generated these JARs directly from the XSDs, at least that's what this POM seems to be doing: https://github.com/4Science/mvn-repo/blob/master/releases/it/4science/tools-orcid-jaxb/2.0/tools-orcid-jaxb-2.0.pom#L80   Do you have any more documentation on how this was done, so that we could potentially replicate this?  I'm not finding the source repository (with the original XSD files) anywhere, so I'm wondering if this is publicly available or not.

Reply:
Schema code seems to be here:  https://github.com/ORCID/ORCID-Source/tree/master/orcid-model/src/main/resources

Reply:
@tdonohue Yes, your guess is exactly, these artifacts are generated from a source repository that is not publicly available. We have a private git repository for project and we generate the Java code from the XSDs copied and pasted form the official master of orcid as pointed by @mwoodiupui (I choose only for https://github.com/ORCID/ORCID-Source/blob/master/orcid-model/src/main/resources/common_2.0/common-2.0.xsd and https://github.com/ORCID/ORCID-Source/tree/master/orcid-model/src/main/resources/record_2.0)

Reply:
@benbosman and/or @jonas-atmire ,

Today, I've updated the new `orcid-jaxb-api` project to use packages/classnames that align pretty well with the ones added by this PR.  See https://github.com/DSpace/orcid-jaxb-api/pull/1

I'd appreciate if one of you could do the following:
* Download & build that PR (`mvn -U clean package`).  This will build a `orcid-jaxb-api.jar`
* Remove all the `org.orcid.*` source code from this PR, and attempt to use that JAR as a replacement.  This may require minor updates to this PR to align with any changed packages/classnames in `orcid-jaxb-api`.
* More details on how to test are in the description of https://github.com/DSpace/orcid-jaxb-api/pull/1

If we find that classes are missing from the generated `orcid-jaxb-api`, please let me know, so that I can work to update that module with what you need for this PR.  Thanks!

Reply:
Currently, I've already built and added the orcid-jaxb dependency locally and removed the previously generated classes.
There were some minor changes to the generated classes, but this was mostly class-names having changed.

The code can already be built, but I haven't had the time yet to check out if the resulting response during a call is actually unmarshallable by these new classes

Reply:
@jonas-atmire and @benbosman : To try and move this discussion along (as it seems to have stalled), I've created an alternate version of this PR which uses `DSpace/orcid-jaxb-api`: https://github.com/DSpace/DSpace/pull/2039   I'd appreciate feedback/testing on this new PR, to see if it better aligns the ideas of this PR with a solution that keeps the ORCID generated API separate.